### PR TITLE
Add implementation dg

### PR DESCRIPTION
**Pull request by ChuangZheQuan**
Screenshot 2021-10-31 at 5 31 35 PM

Delete Appointment Activity Diagram

Screenshot 2021-10-31 at 6 38 47 PM

Delete Appointment Sequence Diagram

Screenshot 2021-10-31 at 5 34 54 PM

Add Appointment Activity Diagram

Screenshot 2021-10-31 at 6 39 07 PM

Add Appointment Sequence Diagram

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -447,6 +447,7 @@ Format: `appt -a p/INDEX_OF_PATIENT d/INDEX_OF_DOCTOR s/DATE_AND_TIME [dur/DURAT
 
 * Both patient's and doctor's index **must be a positive integer** 1, 2, 3, …​
 * `DATE_AND_TIME` accepts the format `DD/MM/YYYY HH:MM` e.g. `20/07/2022`.
+* Duration has to be an integer between 0 and 120 (minutes).
```
Comment (pualixue) [inline docs/UserGuide.md:450]: The current implementation is 1-120 minutes.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -204,8 +204,31 @@ The Sequence Diagram below illustrates the interactions within the Model compone
...
+2. `AddAppointmentCommandParser` will check if the prefixes are relevant. If the prefixes are relevant, a new `AddAppointmentCommand`
+   is created. The `AddAppointmentCommand` implements the `execute()` method from the abstract class `Command`. If the prefixes are not relevant,
+   a `ParseException` will be thrown, and the missing prefixes' error message will be shown.
+3. The `AddAppointmentCommand` will run `execute()`. First, it retrieves the `Patient` object and the `Doctor` object at the given index from the model. If the index is out of range of the
+   `Model#FilteredPatientList` or `Model#FilteredDoctorList`, it will throw an error, and the invalid index message will be shown.
+   If not, the `Appointment` object with the is created and added to the model. The add appointment success message is then returned.
+4. The UI will then display the result
 ### Deleting an appointment
+Deleting an appointment requires the user to input a valid index of the desired appointment in the appointment list.
+####Implementation
```
Comment (pualixue) [inline docs/DeveloperGuide.md:220]: Maybe add a space after `####` to avoid formatting issues.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -204,8 +204,31 @@ The Sequence Diagram below illustrates the interactions within the Model compone
...
+1. After user enters the delete appointment command `appt -d` with an index, the input will be sent
+   to `DeleteAppointmentCommandParser` for parsing.
+2. `DeleteAppointmentCommandParser` will check if the index is valid. If the index is valid, a new `DeleteAppointmentCommand` which extends
```
Comment (pualixue) [inline docs/DeveloperGuide.md:223]: `DeleteAppointmentCommand` which extends `AppointmentCommand` <- missing

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -677,8 +700,71 @@ testers are expected to do more *exploratory* testing.
...
+    1. Prerequisites: list all appointments using the `appt -l` command. There must be multiple doctors and patients in the patient and doctor lists. There are less than 100 patients and doctors. 
+       
+    1. Test case: `appt -a p/1 d/1 s/31/12/2050 12:00 dur/5 r/Patient wants a blood test`<br>
+      Expected: An appointment is added. Details of the appointment shown in the response box. The appointment does not show up in the appointment list.
+
+    1. Test case: `appt -a p/1 d/1 s/31/12/2050 12:05 r/Patient wants a blood test`<br>
+      Expected: An appointment is added. Details of the appointment shown in the response box. The appointment does not show up in the appointment list.
+
+    1. Test case: `appt -a p/1 d/1 s/31/12/2050 12:30 dur/50`<br>
+      Expected: An appointment is added. Details of the appointment shown in the response box. The appointment does not show up in the appointment list.
```
Comment (pualixue) [inline docs/DeveloperGuide.md:713]: Might need to change since we will be showing the added appointment now.

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -677,8 +700,71 @@ testers are expected to do more *exploratory* testing.
 ### Listing all doctors <a name="list-doctors"/>
 
 ### Adding an appointment <a name="appointment"/>
+1. Add an appointment when appointment list is showing today's appointments
+    1. Prerequisites: list all appointments using the `appt -l` command. There must be multiple doctors and patients in the patient and doctor lists. There are less than 100 patients and doctors.
```
Comment (pualixue) [inline docs/DeveloperGuide.md:704]: Perhaps using `appt -f` as they might not have appointments for the current day?

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -677,8 +700,71 @@ testers are expected to do more *exploratory* testing.
...
+      Expected: No appointment is added. Response box displays error message: `The duration should be an integer between 1-120 minutes.`
+
+   1. Test case: `appt -a p/1 d/1 s/30/02/2021 dur/0 r/Patient wants a blood test`<br>
+      Expected: No appointment is added. Response box displays error message: `The duration should be an integer between 1-120 minutes.`
+
 
 ### Deleting an appointment  <a name="delete-appointment"/>
+1. Deleting an appointment while all appointments are being shown
+
+    1. Prerequisites: list all appointments using the `appt -l` command. Multiple appointments in the list.
```
Comment (pualixue) [inline docs/DeveloperGuide.md:746]: Perhaps using `appt -f` as they might not have appointments for the current day?

**Review by pualixue — CHANGES_REQUESTED** [review]
Nice detailed test cases for the manual testing section. Some suggestions to be considered.
For the activity diagrams, I think the if-else conditions need to be in square brackets and an empty diamond.

File: docs/diagrams/AddAppointmentActivityDiagram.puml [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+:User enters delete appointment command with index;
+:Parse command;
+if () then ([required prefixes are present])
+  if () then ([prefix arguments are valid])
+      if () then ([patient index is valid])
+         if () then ([doctor index is valid])
+            : Create new AddAppointmentCommand;
+            : Execute AddAppointmentCommand;
+            : Generate success message;
+         else ([else]])
```
Comment (pualixue) [inline docs/diagrams/AddAppointmentActivityDiagram.puml:14]:
```suggestion
         else ([else])
```

**Review by pualixue — APPROVED** [review]
LGTM

**Review by jiayi1129 — APPROVED** [review]
LGTM! 👍

**Pull request merged**